core/store/src/trie/split_state.rs

```diff
@@ -262,7 +262,7 @@ fn apply_delayed_receipts_to_split_states_impl(
         let trie_update = trie_updates.get_mut(&new_shard_uid).unwrap();
         let trie_key = TrieKey::DelayedReceipt { index: delayed_receipts_indices.first_index };
 
-        let stored_receipt = get::<Receipt>(trie_update, &trie_key)?
+        let stored_receipt = get::<Receipt, _>(trie_update, &trie_key)?
```

Comment:
Yeah, ideally I think all the helper methods should be moved to the new interface rather than being stand alone methods.  (Though Rust not having final kinda irks me if I'm to be honest. ;) )

Reply:
I've heard some rumors about `final` methods in traits, it indeed is bad that Traits today play a dual role of extension point and a provide of convenience methods. Something like this sometimes works, but it is definitely stretching the language :+1: 

```rust
trait Foo {
    fn get(&self, key: &[u8]) -> Vec<u8> { Vec::new() }
}

impl dyn Foo + '_ {
    fn get_ser<T>(&self, key: &[u8]) -> T {
        todo!()
    }
}
```